easybuild/easyconfigs/d/DIAMOND/DIAMOND-2.0.4-GCCcore-9.3.0.eb

```diff
@@ -0,0 +1,28 @@
+easyblock = 'CMakeMake'
+
+name = 'DIAMOND'
+version = "2.0.4"
+
+homepage = 'https://github.com/bbuchfink/diamond'
+description = "Accelerated BLAST compatible local sequence aligner"
+
+toolchain = {'name': 'GCCcore', 'version': '9.3.0'}
```

Comment:
@mboisson Isn't `GCC` a better choice here, since this is performance sensitive?

Reply:
Well, GCCcore also gets hardware accelerations, doesn't it ? Or is that only a local modification to Compute Canada ?

Reply:
It's more about `GCC` vs other compiler-only toolchains like `iccifort`.

If you pick `GCCcore`, you're sort of implying that performance doesn't matter, and that this installation can be used as a dependency for any toolchain that is based on the same version of `GCCcore`.

Reply:
Well, it's an app, not a library. So, yeah... it can be used with any toolchain.

Reply:
```
find /cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/diamond/2.0.4/ | grep -v "2.0.4/easybuild"
/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/diamond/2.0.4/
/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/diamond/2.0.4/bin
/cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/diamond/2.0.4/bin/diamond
```

Reply:
The question is whether the performance of `DIAMOND` is important (compared to something like `CMake`, where it's not at all), and whether it would make sense to compile `DIAMOND` with `iccifort` rather than with `GCC` (it would, imho).

Reply:
You are making a heavy assumption that iccifort produces faster executables than GCC :P    I find that to be debatable, especially for bioinformatics software which quite often don't compile with Intel at all and are developed with GCC. I don't mind bumping it back up to GCC for upstream, I'll just have to squash it down to GCCcore in our stack

Reply:
@mboisson the question is not which compiler is the fastest or which compiler gives the least issues.
the question is: is this program performance critical. if it is, then it makes sense to allow for an alternative compiler to be used so people can test for themselves if it's faster or not.

Reply:
We take a different strategy, but fair enough, I moved it anyway for this PR.